File: fragments/solana_key_pair/solana_key_pair_utils.ts
import { generateKeyPair, getAddressFromPublicKey } from "@solana/kit";
import { webcrypto } from "node:crypto";

interface CryptoKeyPair {
  publicKey: webcrypto.CryptoKey;
  privateKey: webcrypto.CryptoKey;
}

export const createKeyPair = async () => {
  const keypair = await generateKeyPair();
  return keypair;
};

export const getAddress = async (keypair: CryptoKeyPair) => {
  const address = await getAddressFromPublicKey(keypair.publicKey);
  return address;
};

File: fragments/solana_key_pair/solana_key_pair_utils.go
package solana_key_pair

import (
	"github.com/gagliardetto/solana-go"
)

func CreateKeyPair() (solana.PrivateKey, error) {
	return solana.NewRandomPrivateKey()
}

func GetAddress(keypair solana.PrivateKey) solana.PublicKey {
	return keypair.PublicKey()
}
